test/e2e/framework/pod/resize.go

```diff
@@ -395,15 +395,16 @@ func WaitForPodResizeActuation(ctx context.Context, f *framework.Framework, podC
 				// This is a terminal resize state
 				return false, fmt.Errorf("resize is infeasible")
 			}
-			return pod.Status.Resize == "", nil
+
+			return ExpectPodResized(ctx, f, pod, expectedContainers) == nil, nil
```

Comment:
I think this approach is  going to hide the error message from `ExpectPodResized` on failure, which will make this impossible to debug. You could capture the last error, and make sure it gets logged on failure, but that also means if the resize is "done" but failed validation it will take the full timeout for the test to fail.

We're blocked on ObservedGeneration to really implement this the right way (let's leave a TODO here), but for now I think I'd prefer to just wait on `VerifyPodStatusResources`, and then keep the full `ExpectPodResized` as a separate call. This will still require capturing the last error from `VerifyPodStatusResources`. It might be easier to just relpace this `WaitForPodCondition` with a `gomega.Eventually` .

Reply:
thanks! Updated this to wait for `VerifyPodStatusResources` and left `ExpectPodResized` as its own call. I ended up not seeing a strong reason to replace `WaitForPodCondition` with `gomega.Eventually` but I can change it if you prefer